bin/src/lib.rs

```diff
@@ -677,10 +679,7 @@ pub struct TlsCertConfig {
 #[cfg(feature = "dns-over-rustls")]
 impl TlsCertConfig {
     /// Load a Certificate from the path (with rustls)
-    pub fn load(
-        &self,
-        zone_dir: &Path,
-    ) -> Result<(Vec<CertificateDer<'static>>, PrivateKeyDer<'static>), String> {
+    pub fn load(&self, zone_dir: &Path) -> Result<Arc<SingleCertAndKey>, String> {
```

Comment:
I think I'd prefer for this to return an `Arc<dyn ResolvesServerCert>` directly.

Reply:
I'm not 100% certain that we want `dyn` and not `impl` here but if all we do is use it as `Arc<dyn ...>` then it doesn't matter so I changed it to `dyn`. (I think the advantage of `impl` could be that if we used the type directly, the compiler could resolve the calls to an actual type instead of always falling back to dynamic dispatch)